docs/penguins.nim

```diff
@@ -52,34 +54,34 @@ nbImage(url="images/penguins_map.png",
 nbText: """We do expect weight being correlated to some of the length measures
 (e.g. flipper length) with males being bigger than females.
 
-To plot this we need to remove all `NA`s:
+To plot this we need to remove all `NA` and then classify the points both by the penguins sex as
+well as their species:
 """
 # manage runtime error here!
 #nbCode:
-#  ggplot(df, aes(x="body_mass_g", y="flipper_length_mm", color = "sex", shape="species")) + geom_point() + ggsave("images/penguins_mass_vs_length_with_sex_and_species.png")
-#nbImage(url="images/penguins_mass_vs_length_with_sex_and_species.png", caption="Penguins' mass vs flipper length")
+#  ggplot(df, aes(x="body_mass_g", y="flipper_length_mm", color = "sex", shape="species")) + geom_point() + ggsave("images/penguins_mass_vs_length_with_sex_and_species2.png")
+#nbImage(url="images/penguins_mass_vs_length_with_sex_and_species2.png", caption="Penguins' mass vs flipper length")
 nbCode:
-  let df1 = df.filter(f{c"body_mass_g" != "NA"}) # why c""?
-  ggplot(df1, aes(x="body_mass_g", y="flipper_length_mm", color = "sex")) + geom_point() + ggsave("images/penguins_mass_vs_length_with_sex.png")
+  let df1 = df.filter(f{`body_mass_g` != "NA"}) # c"foo" == `foo` == idx("foo") (accent quoted not usable for columns w/ spaces)
```

Comment:
```suggestion
  let df1 = df.filter(f{`body_mass_g` != "NA"}) # c"foo" == `foo` == idx("foo") (backtick quotes not usable for columns with spaces)
```

out of curiosity: really backtick quotes are not working for columns with spaces? seems odd knowing that they can be used for variable names with spaces, but I imagine there are good reasons for that

Reply:
Yes, it's because of a simple limitation of the Nim AST and how it treats these quotes. 

The problem is the following: Anything in accented quotes (as they are called in the Nim AST), will be added to a "list" of identifier parts. So 
```nim
dumpTree:
  `hello`
```
is 
```nim
  AccQuoted
    Ident "hello"
```
If there's a space in there, it becomes:
```nim
dumpTree:
  `hello world`
  AccQuoted
    Ident "hello"
    Ident "world"
```
Seems fine so far right? We just concat the multiple idents using spaces! Look closer though:
```nim
dumpTree:
  `Energy[keV]`
  `Energy [keV]`
```
which gives us:
```nim
  AccQuoted
    Ident "Energy"
    Ident "["
    Ident "keV"
    Ident "]"
  AccQuoted
    Ident "Energy"
    Ident "["
    Ident "keV"
    Ident "]"
```
see the difference? Exactly, there isn't one. So we lose information about where spaces are in certain contexts (i.e. things between which there are already multiple separate identifiers). 

One could choose one or the other (always space, never space) or do some heuristics, but in the end I simply tell people to not use it for spaces or anything that is more complicated. Think about it for columns made up of `{'a' .. 'Z', '0' .. '9'}` and you're good. The `c""` notation was my first attempt to allow for a fallback. Since rewriting a big chunk in datamancer I decided though to go for a more specific `idx("foo")` approach for general string based things (`idx` can also take a string variable for instance. Main reason was to also provide a distinction between referring to a single element of a column (via `idx` which we loop over) or a *full column* using `col("foo")`.